api/src/main/java/com/velocitypowered/api/event/EventTask.java

```diff
@@ -101,13 +55,14 @@ public boolean requiresAsync() {
    * @param task The task
    * @return The async event task
    */
-  public static EventTask.Basic async(final Runnable task) {
+  static EventTask async(final Runnable task) {
     requireNonNull(task, "task");
-    return new Basic() {
+    return new EventTask() {
 
       @Override
-      public void run() {
+      public void execute(Continuation continuation) {
         task.run();
+        continuation.resume();
```

Comment:
```suggestion
        try {
          task.run();
        } catch (Throwable throwable) {
          continuation.resumeWithException(throwable);
          return;
        }
        continuation.resume();
```

Reply:
This sin't necessary, if an exception is thrown during an event task execution, it will be resumed automatically.